File: binding/src/libqsizepolicy.cpp
#include <QSizePolicy>
#include <QVariant>
#include <qsizepolicy.h>
#include "libqsizepolicy.h"
#include "libqsizepolicy.hxx"

QSizePolicy* QSizePolicy_new(const QSizePolicy* other) {
	 return new QSizePolicy(*other);
}

QSizePolicy* QSizePolicy_new2(QSizePolicy* other) {
	 return new QSizePolicy(*other);
}

QSizePolicy* QSizePolicy_new3() {
	 return new QSizePolicy();
}

QSizePolicy* QSizePolicy_new4(int horizontal, int vertical) {
	 return new QSizePolicy(static_cast<QSizePolicy::Policy>(horizontal), static_cast<QSizePolicy::Policy>(vertical));
}

QSizePolicy* QSizePolicy_new5(const QSizePolicy* param1) {
	 return new QSizePolicy(*param1);
}

QSizePolicy* QSizePolicy_new6(int horizontal, int vertical, int typeVal) {
	 return new QSizePolicy(static_cast<QSizePolicy::Policy>(horizontal), static_cast<QSizePolicy::Policy>(vertical), static_cast<QSizePolicy::ControlType>(typeVal));
}

void QSizePolicy_CopyAssign(QSizePolicy* self, QSizePolicy* other) {
    *self = *other;
}

void QSizePolicy_MoveAssign(QSizePolicy* self, QSizePolicy* other) {
    *self = std::move(*other);
}

int QSizePolicy_HorizontalPolicy(const QSizePolicy* self) {
	return self->horizontalPolicy();
}

int QSizePolicy_VerticalPolicy(const QSizePolicy* self) {
	return self->verticalPolicy();
}

int QSizePolicy_ControlType(const QSizePolicy* self) {
	return self->controlType();
}

void QSizePolicy_SetHorizontalPolicy(QSizePolicy* self, int d) {
	self->setHorizontalPolicy(static_cast<QSizePolicy::Policy>(d));
}

void QSizePolicy_SetVerticalPolicy(QSizePolicy* self, int d) {
	self->setVerticalPolicy(static_cast<QSizePolicy::Policy>(d));
}

void QSizePolicy_SetControlType(QSizePolicy* self, int typeVal) {
	self->setControlType(static_cast<QSizePolicy::ControlType>(typeVal));
}

int QSizePolicy_ExpandingDirections(const QSizePolicy* self) {
	return self->expandingDirections();
}

void QSizePolicy_SetHeightForWidth(QSizePolicy* self, bool b) {
	self->setHeightForWidth(b);
}

bool QSizePolicy_HasHeightForWidth(const QSizePolicy* self) {
	return self->hasHeightForWidth();
}

void QSizePolicy_SetWidthForHeight(QSizePolicy* self, bool b) {
	self->setWidthForHeight(b);
}

bool QSizePolicy_HasWidthForHeight(const QSizePolicy* self) {
	return self->hasWidthForHeight();
}

bool QSizePolicy_OperatorEqual(const QSizePolicy* self, const QSizePolicy* s) {
	return self->operator==(*s);
}

bool QSizePolicy_OperatorNotEqual(const QSizePolicy* self, const QSizePolicy* s) {
	return self->operator!=(*s);
}

QVariant* QSizePolicy_OperatorQVariant(const QSizePolicy* self) {
	return new QVariant(self->operator QVariant());
}

int QSizePolicy_HorizontalStretch(const QSizePolicy* self) {
	return self->horizontalStretch();
}

int QSizePolicy_VerticalStretch(const QSizePolicy* self) {
	return self->verticalStretch();
}

void QSizePolicy_SetHorizontalStretch(QSizePolicy* self, int stretchFactor) {
	self->setHorizontalStretch(stretchFactor);
}

void QSizePolicy_SetVerticalStretch(QSizePolicy* self, int stretchFactor) {
	self->setVerticalStretch(stretchFactor);
}

bool QSizePolicy_RetainSizeWhenHidden(const QSizePolicy* self) {
	return self->retainSizeWhenHidden();
}

void QSizePolicy_SetRetainSizeWhenHidden(QSizePolicy* self, bool retainSize) {
	self->setRetainSizeWhenHidden(retainSize);
}

void QSizePolicy_Transpose(QSizePolicy* self) {
	self->transpose();
}

QSizePolicy* QSizePolicy_Transposed(const QSizePolicy* self) {
	return new QSizePolicy(self->transposed());
}

void QSizePolicy_Delete(QSizePolicy* self) {
    delete self;
}

File: binding/include/libqsizepolicy.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qsizepolicy.html
pub const qsizepolicy = struct {

    /// New constructs a new QSizePolicy object.
    pub fn New(other: ?*anyopaque) QtC.QSizePolicy {
        return qtc.QSizePolicy_new(@ptrCast(other));
    }


    /// New2 constructs a new QSizePolicy object.
    pub fn New2(other: ?*anyopaque) QtC.QSizePolicy {
        return qtc.QSizePolicy_new2(@ptrCast(other));
    }


    /// New3 constructs a new QSizePolicy object.
    pub fn New3() QtC.QSizePolicy {
        return qtc.QSizePolicy_new3();
    }


    /// New4 constructs a new QSizePolicy object.
    pub fn New4(horizontal: i32, vertical: i32) QtC.QSizePolicy {
        return qtc.QSizePolicy_new4(@intCast(horizontal), @intCast(vertical));
    }


    /// New5 constructs a new QSizePolicy object.
    pub fn New5(param1: ?*anyopaque) QtC.QSizePolicy {
        return qtc.QSizePolicy_new5(@ptrCast(param1));
    }


    /// New6 constructs a new QSizePolicy object.
    pub fn New6(horizontal: i32, vertical: i32, typeVal: i32) QtC.QSizePolicy {
        return qtc.QSizePolicy_new6(@intCast(horizontal), @intCast(vertical), @intCast(typeVal));
    }

    /// CopyAssign shallow copies `other` into `self`.
    pub fn CopyAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QSizePolicy_CopyAssign(@ptrCast(self), @ptrCast(other));
    }

    /// MoveAssign moves `other` into `self` and invalidates `other`.
    pub fn MoveAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QSizePolicy_MoveAssign(@ptrCast(self), @ptrCast(other));
    }


    pub fn HorizontalPolicy(self: ?*anyopaque, ) i32 {
        return qtc.QSizePolicy_HorizontalPolicy(@ptrCast(self));
    }

    pub fn VerticalPolicy(self: ?*anyopaque, ) i32 {
        return qtc.QSizePolicy_VerticalPolicy(@ptrCast(self));
    }

    pub fn ControlType(self: ?*anyopaque, ) i32 {
        return qtc.QSizePolicy_ControlType(@ptrCast(self));
    }

    pub fn SetHorizontalPolicy(self: ?*anyopaque, d: i32) void {
        qtc.QSizePolicy_SetHorizontalPolicy(@ptrCast(self), @intCast(d));
    }

    pub fn SetVerticalPolicy(self: ?*anyopaque, d: i32) void {
        qtc.QSizePolicy_SetVerticalPolicy(@ptrCast(self), @intCast(d));
    }

    pub fn SetControlType(self: ?*anyopaque, typeVal: i32) void {
        qtc.QSizePolicy_SetControlType(@ptrCast(self), @intCast(typeVal));
    }

    pub fn ExpandingDirections(self: ?*anyopaque, ) i32 {
        return qtc.QSizePolicy_ExpandingDirections(@ptrCast(self));
    }

    pub fn SetHeightForWidth(self: ?*anyopaque, b: bool) void {
        qtc.QSizePolicy_SetHeightForWidth(@ptrCast(self), b);
    }

    pub fn HasHeightForWidth(self: ?*anyopaque, ) bool {
        return qtc.QSizePolicy_HasHeightForWidth(@ptrCast(self));
    }

    pub fn SetWidthForHeight(self: ?*anyopaque, b: bool) void {
        qtc.QSizePolicy_SetWidthForHeight(@ptrCast(self), b);
    }

    pub fn HasWidthForHeight(self: ?*anyopaque, ) bool {
        return qtc.QSizePolicy_HasWidthForHeight(@ptrCast(self));
    }

    pub fn OperatorEqual(self: ?*anyopaque, s: ?*anyopaque) bool {
        return qtc.QSizePolicy_OperatorEqual(@ptrCast(self), @ptrCast(s));
    }

    pub fn OperatorNotEqual(self: ?*anyopaque, s: ?*anyopaque) bool {
        return qtc.QSizePolicy_OperatorNotEqual(@ptrCast(self), @ptrCast(s));
    }

    pub fn OperatorQVariant(self: ?*anyopaque, ) QtC.QVariant {
        return qtc.QSizePolicy_OperatorQVariant(@ptrCast(self));
    }

    pub fn HorizontalStretch(self: ?*anyopaque, ) i32 {
        return qtc.QSizePolicy_HorizontalStretch(@ptrCast(self));
    }

    pub fn VerticalStretch(self: ?*anyopaque, ) i32 {
        return qtc.QSizePolicy_VerticalStretch(@ptrCast(self));
    }

    pub fn SetHorizontalStretch(self: ?*anyopaque, stretchFactor: i32) void {
        qtc.QSizePolicy_SetHorizontalStretch(@ptrCast(self), stretchFactor);
    }

    pub fn SetVerticalStretch(self: ?*anyopaque, stretchFactor: i32) void {
        qtc.QSizePolicy_SetVerticalStretch(@ptrCast(self), stretchFactor);
    }

    pub fn RetainSizeWhenHidden(self: ?*anyopaque, ) bool {
        return qtc.QSizePolicy_RetainSizeWhenHidden(@ptrCast(self));
    }

    pub fn SetRetainSizeWhenHidden(self: ?*anyopaque, retainSize: bool) void {
        qtc.QSizePolicy_SetRetainSizeWhenHidden(@ptrCast(self), retainSize);
    }

    pub fn Transpose(self: ?*anyopaque, ) void {
        qtc.QSizePolicy_Transpose(@ptrCast(self));
    }

    pub fn Transposed(self: ?*anyopaque, ) QtC.QSizePolicy {
        return qtc.QSizePolicy_Transposed(@ptrCast(self));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QSizePolicy_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/sizepolicy.html#types
pub const enums = struct {
    pub const PolicyFlag = enum {
        pub const GrowFlag: i32 = 1;
        pub const ExpandFlag: i32 = 2;
        pub const ShrinkFlag: i32 = 4;
        pub const IgnoreFlag: i32 = 8;
    };

    pub const Policy = enum {
        pub const Fixed: i32 = 0;
        pub const Minimum: i32 = 1;
        pub const Maximum: i32 = 4;
        pub const Preferred: i32 = 5;
        pub const MinimumExpanding: i32 = 3;
        pub const Expanding: i32 = 7;
        pub const Ignored: i32 = 13;
    };

    pub const ControlType = enum {
        pub const DefaultType: i32 = 1;
        pub const ButtonBox: i32 = 2;
        pub const CheckBox: i32 = 4;
        pub const ComboBox: i32 = 8;
        pub const Frame: i32 = 16;
        pub const GroupBox: i32 = 32;
        pub const Label: i32 = 64;
        pub const Line: i32 = 128;
        pub const LineEdit: i32 = 256;
        pub const PushButton: i32 = 512;
        pub const RadioButton: i32 = 1024;
        pub const Slider: i32 = 2048;
        pub const SpinBox: i32 = 4096;
        pub const TabWidget: i32 = 8192;
        pub const ToolButton: i32 = 16384;
    };

};
